src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -8008,47 +7981,42 @@ export class StudyViewPageStore
     readonly molecularProfileSampleCounts = remoteData<
         MultiSelectionTableRow[]
     >({
-        await: () => [this.molecularProfiles, this.selectedSamples],
+        await: () => [this.molecularProfiles],
         invoke: async () => {
-            // return empty if there are no filtered samples
-            if (!this.hasFilteredSamples) {
-                return [];
-            }
-            const counts = await internalClient.fetchMolecularProfileSampleCountsUsingPOST(
-                {
+            const [counts, selectedSamples] = await Promise.all([
+                internalClient.fetchMolecularProfileSampleCountsUsingPOST({
                     studyViewFilter: this.filters,
-                }
-            );
+                }),
+                toPromise(this.selectedSamples),
```

Comment:
this is sort of the heart of this PR here, right?  We need to use all so that we can concurrently load selectedSamples

Reply:
what's kinda weird though is that selectedSamples itself depends on filters.  maybe it's not weird.

Reply:
in this case, yes - in other cases, we can just remove the selectedSamples dependency completely